lib/zwave/ZwaveDevice.js

```diff
@@ -753,26 +753,40 @@ class ZwaveDevice extends MeshDevice {
 		commandClass.on('report', (command, payload) => {
 			if (command.name !== commandId) return;
 
- 			// Catch central scene echos and (sometimes) failing parser
- 			if (command.name === 'CENTRAL_SCENE_NOTIFICATION') {
-
- 				if (typeof previousSequence !== 'undefined' && payload.hasOwnProperty('Sequence Number') && payload['Sequence Number'] === previousSequence) return;
- 				previousSequence = payload['Sequence Number'];
-
- 				if (payload.hasOwnProperty('Properties1') &&
- 					payload.Properties1.hasOwnProperty('Key Attributes') &&
- 					typeof payload.Properties1['Key Attributes'] === 'number') {
- 					switch(payload.Properties1['Key Attributes']) {
- 						case 0: payload.Properties1['Key Attributes'] = 'Key Pressed 1 time'; break;
- 						case 1: payload.Properties1['Key Attributes'] = 'Key Released'; break;
- 						case 2: payload.Properties1['Key Attributes'] = 'Key Held Down'; break;
- 						case 3: payload.Properties1['Key Attributes'] = 'Key Pressed 2 times'; break;
- 						case 4: payload.Properties1['Key Attributes'] = 'Key Pressed 3 times'; break;
- 						case 5: payload.Properties1['Key Attributes'] = 'Key Pressed 4 times'; break;
- 						case 6: payload.Properties1['Key Attributes'] = 'Key Pressed 5 times'; break;
- 					}
- 				}
- 			}
+			// Catch central scene echos and (sometimes) failing parser
+			if (command.name === 'CENTRAL_SCENE_NOTIFICATION') {
```

Comment:
Why not put this in a specific Command Class file? `/lib/zwave/system/commandclasses/`

Reply:
Cause that didn't exist when I added this few months ago :smile:

Reply:
Fair enough. Can you split it up? I think command class specific code doesn't belong in the base class.

Reply:
The command class code wasn't added with this commit (only its style was fixed). I suggest to merge this PR and refactor the central scene parsers as suggested by @WeeJeWel in another PR.